src/domain/SystemEnv.ts

```diff
@@ -0,0 +1,22 @@
+import { IO } from 'fp-ts/IO';
+import { Iun } from '../generated/definitions/Iun';
+import { CheckNotificationStatusRecordRepository } from './CheckNotificationStatusRepository';
+import { ConsumeEventStreamRecordRepository } from './ConsumeEventStreamRecordRepository';
+import { GetNotificationDetailRecordRepository } from './GetNotificationDetailRepository';
+import { GetNotificationDocumentMetadataRecordRepository } from './GetNotificationDocumentMetadataRepository';
+import { NewNotificationRepository } from './NewNotificationRepository';
+
+export type SystemEnv = {
+  // envs
+  occurrencesAfterComplete: number;
+  senderPAId: string;
+  // generators
+  iunGenerator: IO<Iun>;
+  dateGenerator: IO<Date>;
+  // repositories
+  createNotificationRequestRecordRepository: NewNotificationRepository;
+  findNotificationRequestRecordRepository: CheckNotificationStatusRecordRepository;
+  consumeEventStreamRecordRepository: ConsumeEventStreamRecordRepository;
+  getNotificationDetailRecordRepository: GetNotificationDetailRecordRepository;
+  getNotificationDocumentMetadataRecordRepository: GetNotificationDocumentMetadataRecordRepository;
```

Comment:
Just a couple of questions about this:
why the repositories should be a part of the system environment?
And why the system env does not contain all the repositories? (e.g. the PreLoadRepository, ecc.)

Reply:
> Just a couple of questions about this: why the repositories should be a part of the system environment? 

The `SystemEnv` is born to keep some common dependencies on it, but I think neither the name nor the place are the best ones. Here are two options that should improve comprehension: 
1. Move `SystemEnv` from `domain` folder to `useCases` folder.
   - [Optional] Rename  `SystemEnv` to `UseCasesEnv`.
2. Split the `SystemEnv` into two `Env`, one for the domain (the `DomainEnv`) and one for the useCases (the `UseCasesEnv`), the latter will contain repositories and the `DomainEnv`.

> And why the system env does not contain all the repositories? (e.g. the PreLoadRepository, ecc.)

The goal of this refactor is to remove some code duplication, the repositories contained in this structure are those used in the use cases involved in this refactor.

Reply:
Yeah, this work simplifies a lot the code, but speaking of _responsibility_, I think it's better to split the `SystemEnv` into two separate envs with the dependency you said.

Reply:
Done 🍰